### DISCUSSION TITLE
Enable agent auto upgrade with runner scale sets

### DISCUSSION THREAD

**Discussion by chirauki** [community/community#160711] [Actions]
Hello,

Using [gha-runner-scale-set](https://github.com/actions/actions-runner-controller/tree/master/charts/gha-runner-scale-set) chart, the resulting ephemeral runners have `runnerJITConfig` populated, which includes:

```json
{
  "AgentId": "xxxx",
  "AgentName": "xxx",
  "DisableUpdate": "True",
  "Ephemeral": "True",
  "PoolId": "1",
  "PoolName": null,
  "ServerUrl": "https://pipelinesghubeus22.actions.githubusercontent.com/xxxx/",
  "RunnerScaleSetId": "3",
  "RunnerScaleSetName": "default-runner-set",
  "WorkFolder": "_work"
}
```

Note `"DisableUpdate": "True"`, which disables auto update on runner start.

I couldn't find a way to change this in the docs... Do you know how this setting can be changed?

**Comment by Constantin07** [comment]
I'm also having similar issue:

```
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO ConfigurationManager] Settings Loaded
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener] {
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "AgentId": 9163,
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "AgentName": "arc-runner-playwright-vq7wn-runner-l8xh7",
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "PoolId": 1,
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "DisableUpdate": true,
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "Ephemeral": true,
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "ServerUrl": "https://pipelinesghubeus4.actions.githubusercontent.com/*****/",
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener]   "WorkFolder": "_work"
arc-runner-playwright-vq7wn-runner-l8xh7 runner [RUNNER 2024-05-17 12:52:18Z INFO MessageListener] } 
```

This https://github.com/actions/actions-runner-controller/blob/3bda9bb2402b361e3bab78fdcd2b67da1c4e3638/runner/startup.sh#L86 suggests that it can be disabled (by default enabled ?) but setting the `DISABLE_RUNNER_UPDATE` to `false` or not defining it has no effect - updates still disabled.

Would be nice to have some clarification.

**Reply by air-hand** [reply]
It seems impossible.
https://github.com/actions/actions-runner-controller/blob/master/controllers/actions.github.com/autoscalingrunnerset_controller.go#L461
IMO, The quickest way is to use `ghcr.io/actions/actions-runner:latest` image.
